numpy/core/arrayprint.py

```diff
@@ -704,8 +710,8 @@ def fillFormat(self, data):
         with errstate(all='ignore'):
             hasinf = isinf(data)
             special = isnan(data) | hasinf
-            valid = not_equal(data, 0) & ~special
-            non_zero = data[valid]
+            non_special = data[~special]
+            non_zero = non_special[non_special != 0]
```

Comment:
Some comments might be nice here explaining why each of these are needed.

Reply:
A 'not' prefix has more force than 'non'. Not worth worrying about here, just a comment.

Reply:
I would reserve a `not_` prefix for a boolean type, so `non_zero = data[not_zero]`